git/remote.py

```diff
@@ -116,6 +116,23 @@ def to_progress_instance(progress: Union[Callable[..., Any], RemoteProgress, Non
     return progress
 
 
+class PushInfoList(IterableList):
```

Comment:
The strange thing is that I would expect `IterableList[PushInfo]` to be the base class, or at least something in this type should indicate that `PushInfo`s are iterated over.

That way, `PushInfoList` would be fitting in place of `IterableList[PushInfo]`, which might even be a test to add to assure the type system stays compatible, too.

Something like…

```python

def f(l: IterableList[PushInfo]):
    pass

f(PushInfoList())
```

Reply:
I agree, but I had trouble [getting this to work with mypy](https://github.com/gitpython-developers/GitPython/runs/4165677582?check_suite_focus=true).

The list with pushinfos was created like this:

```
output: IterableList[PushInfo] = IterableList('push_infos')
```

I am not sure what the `push_infos` argument is for, but I tried moving it into PushInfoList:

```
class PushInfoList(IterableList[PushInfo]):
    def __new__(cls) -> 'PushInfoList':
        return super().__new__(cls, 'push_infos')

    def __init__(self) -> None:
        super().__init__('push_infos')
        self.error: Optional[Exception] = None
```

However, mypy doesn't agree with this. The types don't match up in `__new__`. `super(IterableList[PushInfo], cls)` doesn't work, so now I have a call explicitly to `IterableList.__new__`. Is that OK?

```
class PushInfoList(IterableList[PushInfo]):
    def __new__(cls) -> 'PushInfoList':
        return cast(PushInfoList, IterableList.__new__(cls, 'push_infos'))
```

Reply:
`"push_infos"` is supposed to be an `id_attribute`, which definitely isn't useful here as [it is used like this](https://github.com/gitpython-developers/GitPython/blob/main/git/util.py#L995:L1001).

Maybe @Yobmod can help us clarify if this `.__new__` calling convention to pacify mypi is the expected way, or if there are more idiomatic ways to do this.

From my point of view, it's certainly OK even though I can't tell if a future patch release of mypi is going to stop accepting it.